pyccel/codegen/printing/fcode.py

```diff
@@ -2608,31 +2608,77 @@ def _print_IndexedElement(self, expr):
                     shape = var.shape, prec = var.precision,
                     order = var.order, rank = var.rank)[expr.indices[:-1]])
         else:
-            base = self._print(expr.base.label)
+            base_code = self._print(expr.base.label)
 
         inds = list(expr.indices)
         base_shape = Shape(expr.base)
         allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
                 expr.base.internal_variable.allows_negative_indexes)
 
         for i, ind in enumerate(inds):
-            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], LiteralInteger):
-                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            _shape = PyccelArraySize(base, i if expr.order != 'C' else len(inds) - i - 1)
+            if isinstance(ind, Slice):
+                inds[i] = self._slice_management(ind, _shape, allow_negative_indexes)
+            elif isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], LiteralInteger):
+                inds[i] = PyccelMinus(_shape, ind.args[0])
             else:
                 #indices of indexedElement of len==1 shouldn't be a Tuple
                 if isinstance(ind, Tuple) and len(ind) == 1:
                     inds[i] = ind[0]
                 if allow_negative_indexes and not isinstance(ind, LiteralInteger):
-                    inds[i] = PyccelMod(ind, base_shape[i])
+                    inds[i] = PyccelMod(ind, _shape)
 
         inds = [self._print(i) for i in inds]
 
-        return "%s(%s)" % (base, ", ".join(inds))
+        return "%s(%s)" % (base_code, ", ".join(inds))
 
 
     def _print_Idx(self, expr):
         return self._print(expr.label)
 
+    def _slice_management(self, _slice, shape, allow_negative_index):
```

Comment:
Is there a difference between this method and the one in the ccode? If not then I wonder if it would make sense to make it a member of the slice class. For example we could do:
`slice = slice.remove_negative_indexes(shape)` for example?

What do you think @saidctb @yguclu ?

Reply:
Or `slice.start_w_pos_indexes`, `slice.step_w_pos_indexes`, and `slice.end_w_pos_indexes` ?

Reply:
slice management in fcode is different than the one in ccode. the management of slice in c code is done like the one in python. but Fortran had some differences.